### DISCUSSION TITLE
Invest and unit commitment issue

### DISCUSSION THREAD

**Discussion by tokorte** [spine-tools/SpineOpt.jl#858] [Support]
We are currently working on our Spine Model, but we ran into an issue with invest and unit commitment using a combination of continuous invest and integer unit commitment. Maybe you can help us solve the issue or lead us in the right direction:

In short, what we want to model:
•            Investment: A unit that can be invested in (continuously) with a max. investible capacity of 150 MW. 
•            Unit Commitment: The unit should either be offline or run at 100% of the invested capacity (e.g., run at 70 MW if 70 MW is the invested capacity). 

These parameters (excerpt) were set in the model for the respective unit:
•            `candidate_units` = 1  
•            `unit_capacity` = 150 MW
•            `unit_investment_variable_type` = continuous
•            `online_variable_type` = integer

With this setup, we can observe an integer unit commitment as planned, but the unit investment decision is also integer. I.e., we don’t observe a continuous investment decision ranging from 0-150 MW, but instead integer investment in a 0 or 150 MW unit is observable. Is this the expected behaviour and if not, how can we solve the issue? 

Also, I would like to push an Issue #715 we observed regarding investment in connections. Thanks in advance.

**Comment by DillonJ** [comment]
Hi @tokorte 

A quick question, based on your experiments, are you sure it's not working correctly? You are making a conclusion based on a solution for your investment decision... but are you sure that this is not simply the result? To make a conclusions, you would have to create an example where you know the optimal result is a non-integer investment decision

**Comment by DillonJ** [comment]
Actually, come to think of it... by enforcing integrality on the commitment variable, you effectively enforce integrality on the investment decision because even if the invested_units is 0.5, your units_on must be 0 or 1.

**Comment by NEnglisch** [comment]
Hi @DillonJ, I work together with @tokorte on this project.

We conducted several simulations with varying parameters such as investment costs and fuel costs, and analyzed the transition from an investment to no investment phase. The results indicated that there was only an integer investment decision. We also ran simulations with the linear online_variable_type, which showed that there are non-integer investment decisions. The problem with the linear online type is that our unit starts to run at part load. Because of that, we chose the integer online type to ensure that the unit always runs at 100% or is offline. Your example illustrates the scenario we aim for, where the number of invested units ranges from 0 to 1, and the units_on variable is either 0 or 1.

**Comment by DillonJ** [comment]
@tokorte the solution in your example to getting a continuous investment decision is my making the unit_capacity = 1 and candidate_units = 150. However, if the investment decisions was units_invested_available = 100, then this would, in fact be 100 units of 1MW capacity, so in any time interval, units_on would be any integer beween 0 and 100 which is probably not what you want either.

**Comment by tokorte** [comment]
Hello Dillon, thank you for reaching out.

> units_on would be any integer beween 0 and 100 which is probably not what you want either.

Yes, this is not what we would want. We would want it to be 0 or 100, but nothing in between.

We also tried to introduce a `min_operating_point` = 1 for the unit (along with a linear online type and candiate_units = 1). But the `min_operating_point` constraints `online units`, and with a linear online type a unit can be partially shut down, so this constraint seems to be without effect in this case. Would there be a scenario in which `min_operating_point` =1 in combination with continuous invest and linear online type would work? We also created several custom constraints (based on the `min_operating_point` constraint) to test different behaviours of the unit.

**Comment by clizbe** [comment]
@tokorte Were you able to resolve this issue?

**Reply by tokorte** [reply]
Hello @clizbe, thanks for checking in. We are currently trying to implement a different approach to solve this issue. When we have updates, we will report back.
